internal/build/container_ops.go

```diff
@@ -138,6 +140,53 @@ func findMount(info types.ContainerJSON, dst string) (types.MountPoint, error) {
 	return types.MountPoint{}, fmt.Errorf("no matching mount found for %s", dst)
 }
 
+//WriteProjectMetadata
+func WriteProjectMetadata(p string, metadata platform.ProjectMetadata, os string) ContainerOperation {
+	return func(ctrClient client.CommonAPIClient, ctx context.Context, containerID string, stdout, stderr io.Writer) error {
+		buf := &bytes.Buffer{}
+		err := toml.NewEncoder(buf).Encode(metadata)
+		if err != nil {
+			return errors.Wrap(err, "marshaling project metadata")
+		}
+
+		tarBuilder := archive.TarBuilder{}
+
+		tarPath := p
+		if os == "windows" {
+			tarPath = paths.WindowsToSlash(p)
+		}
+
+		tarBuilder.AddFile(tarPath, 0755, archive.NormalizedDateTime, buf.Bytes())
+		reader := tarBuilder.Reader(archive.DefaultTarWriterFactory())
+		defer reader.Close()
+
+		if os == "windows" {
+			dirName := paths.WindowsDir(p)
+			return copyDirWindows(ctx, ctrClient, containerID, reader, dirName, stdout, stderr)
+		}
+
+		return ctrClient.CopyToContainer(ctx, containerID, "/", reader, types.CopyToContainerOptions{})
+	}
+}
+
+// func createReader(src, dst string, uid, gid int, includeRoot bool, fileFilter func(string) bool) (io.ReadCloser, error) {
+// 	fi, err := os.Stat(src)
+// 	if err != nil {
+// 		return nil, err
+// 	}
+
+// 	if fi.IsDir() {
+// 		var mode int64 = -1
+// 		if runtime.GOOS == "windows" {
+// 			mode = 0777
+// 		}
+
+// 		return archive.ReadDirAsTar(src, dst, uid, gid, mode, false, includeRoot, fileFilter), nil
+// 	}
+
+// 	return archive.ReadZipAsTar(src, dst, uid, gid, -1, false, fileFilter), nil
+// }
+
```

Comment:
We don't want to commit unused code

```suggestion
```

![image](https://user-images.githubusercontent.com/475559/127559309-fd1c4c91-fca8-4f5e-bf9d-169c55de71df.png)